src/audio/host-zephyr.c

```diff
@@ -957,6 +945,30 @@ static int host_params(struct comp_dev *dev,
 	return err;
 }
 
+/* configure the DMA params and descriptors for host buffer IO */
+static int host_params(struct comp_dev *dev,
+		       struct sof_ipc_stream_params *params)
+{
+	struct host_data *hd = comp_get_drvdata(dev);
+	int err;
+
+	comp_dbg(dev, "host_params()");
+
+	err = host_verify_params(dev, params);
+	if (err < 0) {
+		comp_err(dev, "host_params(): pcm params verification failed.");
+		return -EINVAL;
```

Comment:
why throw away err? Just return err here

Reply:
at that time, I spent lot of time on params function, the only possible value is -EINVAL.
However, you are right, I am changing.